### PR TITLE
Expose more functions of ngql

### PR DESCRIPTION
**Pull request by arv**
This exports a few functions so that we can pass variables into
the query:

 - NewRootQueryObject (was constructQueryType)
 - GetTypeName
 - GetInputTypeName
 - InputToNomsValue (was argToNomsValue)

### CODE REVIEW

File: go/ngql/types.go [hunk 1]
```diff
@@ -860,7 +871,7 @@ func MaybeGetScalar(v types.Value) interface{} {
 	return v
 }
 
-func argToNomsValue(arg interface{}, nomsType *types.Type) types.Value {
+func InputToNomsValue(arg interface{}, nomsType *types.Type) types.Value {
```
Comment (kalman) [inline go/ngql/types.go]: godoc

File: go/ngql/query_test.go [hunk 2]
```diff
@@ -1114,3 +1115,74 @@ func (suite *QueryGraphQLSuite) TestErrorsInInputType() {
 		"l": types.MakeListType(types.MakeCycleType(0)),
 	}))
 }
+
+func (suite *QueryGraphQLSuite) TestVariables() {
```
Comment (kalman) [inline go/ngql/query_test.go]: this seems unrelated?

Comment (arv) [inline go/ngql/query_test.go]: I wanted to make sure all the right pieces were exposed and that variables use the same kind of conversions.

**Comment by arv** [comment]
jenkins: test this
